lib/cipher.rb

```diff
@@ -33,4 +33,52 @@ def shift(char, shift)
       char
     end
   end
+
+  def crack_key(message)
+    shift_labels = %w(0 1 2 3).rotate(message.length % 4)
+    shifts = []
+    shifts << ALPHA.index(message[-4]) - ALPHA.index(" ")
+    shifts << ALPHA.index(message[-3]) - ALPHA.index("e")
+    shifts << ALPHA.index(message[-2]) - ALPHA.index("n")
+    shifts << ALPHA.index(message[-1]) - ALPHA.index("d")
+
+    shift_labels.zip(shifts).to_h
+    # return is a hash w shift values to decrypt, using keys 0-3 as a-d
+  end
+
+  def crack_chars(message)
+    shift_hash = crack_key(message)
+    prepare(message).reduce([]) do |acc, char|
+        acc << shift(char[0], -shift_hash["0"])
+        acc << shift(char[1], -shift_hash["1"]) if char[1]
+        acc << shift(char[2], -shift_hash["2"]) if char[2]
+        acc << shift(char[3], -shift_hash["3"]) if char[3]
+        acc
+    end
+  end
+
+  # def crack_key_string(message, date)
```

Comment:
This big commented out method is me _trying_ to explore how to return an actual key string to no avail.